sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -179,28 +200,14 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
       throw error;
     }
   }
-
-  private async _createMessageSessionIfDoesntExist(): Promise<void> {
-    // TODO - pass timeout for MessageSession creation
-    if (this._messageSession) {
-      return;
-    }
-    this._context.isSessionEnabled = true;
-    this._messageSession = await MessageSession.create(this._context, {
-      sessionId: this._sessionOptions.sessionId,
-      autoRenewLockDurationInMs: this._sessionOptions.autoRenewLockDurationInMs,
-      receiveMode: convertToInternalReceiveMode(this.receiveMode)
-    });
-    // By this point, we should have a valid sessionId on the messageSession
-    // If not, the receiver cannot be used, so throw error.
-    if (this._messageSession.sessionId == null) {
-      const error = new Error("Something went wrong. Cannot lock a session.");
-      log.error(`[${this._context.namespace.connectionId}] %O`, error);
-      throw error;
+  private _throwIfMessageSessionDoesntExist(): void {
+    if (!this._messageSession) {
```

Comment:
`this._messageSession` will never be undefined. I believe a better check would be `if (!this._context.messageSessions[this.sessionId])`

So, if there has been any issues with the underlying AMQP link, the code inside `MessageSession` class ensures to clear this session from `this._context.messageSessions`

Reply:
> (!this._context.messageSessions[this.sessionId])

Nice.

Yeah, I've noticed and was thinking in the lines of using `this._messageSession.isOpen()`. Should I check both?

Reply:
Added both the checks, I need to see the cases for these code flows (and also add tests.)